[Bug] UpstreamCheckServiceTest.testReplace flaky test due to race condition

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Current Behavior

https://github.com/apache/shenyu/actions/runs/26797371607/job/78996563313?pr=6341

> Error: -admin] [ERROR] Failures: 
> Error: -admin] [ERROR]   UpstreamCheckServiceTest.testReplace:230 expected: &lt;localhost2&gt; but was: &lt;localhost&gt;

```code
Main thread                         Executor thread
───────────────────────             ───────────────────────
submit → submitJust (sync)
  UPSTREAM_MAP = [localhost]
  executor.execute(task) ──→        (task queued)
replace → UPSTREAM_MAP = [localhost2]
assert(localhost2)                  task runs:
  actual: localhost ✗                 updateHandler → UPSTREAM_MAP.put([localhost])
                                       ↑ overwrites replace() result
```


### Expected Behavior

_No response_

### Steps To Reproduce

_No response_

### Environment

```markdown
ShenYu version(s):
```

### Debug logs

_No response_

### Anything else?

_No response_

--- a/shenyu-admin/src/test/java/org/apache/shenyu/admin/service/UpstreamCheckServiceTest.java
+++ b/shenyu-admin/src/test/java/org/apache/shenyu/admin/service/UpstreamCheckServiceTest.java
@@ -221,8 +221,9 @@ public final class UpstreamCheckServiceTest {
         final DivideUpstream divideUpstream2 = DivideUpstream.builder()
                 .upstreamHost("localhost2")
                 .build();
+        upstreamMap.remove(MOCK_SELECTOR_NAME_2);
         upstreamMap.put(MOCK_SELECTOR_NAME_2, new CopyOnWriteArrayList<>(Collections.singletonList(divideUpstream)));
-        upstreamCheckService.replace(MOCK_SELECTOR_NAME_2, Collections.singletonList(divideUpstream2));
+        upstreamCheckService.replace(MOCK_SELECTOR_NAME_2, new CopyOnWriteArrayList<>(Collections.singletonList(divideUpstream2)));
         assertEquals(1, upstreamMap.get(MOCK_SELECTOR_NAME_2).size());
         assertEquals("localhost2", upstreamMap.get(MOCK_SELECTOR_NAME_2).get(0).getUpstreamHost());
     }

--- a/shenyu-admin/src/test/java/org/apache/shenyu/admin/service/UpstreamCheckServiceTest.java
+++ b/shenyu-admin/src/test/java/org/apache/shenyu/admin/service/UpstreamCheckServiceTest.java
@@ -58,6 +58,7 @@ import java.util.List;
 import java.util.Map;
 import java.util.Properties;
 import java.util.Set;
+import java.util.concurrent.CopyOnWriteArrayList;
 import java.util.concurrent.ScheduledThreadPoolExecutor;
 
 import static org.hamcrest.MatcherAssert.assertThat;
@@ -214,17 +215,13 @@ public final class UpstreamCheckServiceTest {
 
     @Test
     public void testReplace() {
-        ScheduledThreadPoolExecutor executor = new ScheduledThreadPoolExecutor(1, ShenyuThreadFactory.create("scheduled-upstream-task", false));
-        ReflectionTestUtils.setField(upstreamCheckService, "executor", executor);
         final DivideUpstream divideUpstream = DivideUpstream.builder()
                 .upstreamHost("localhost")
                 .build();
         final DivideUpstream divideUpstream2 = DivideUpstream.builder()
                 .upstreamHost("localhost2")
                 .build();
-        upstreamCheckService.submit(MOCK_SELECTOR_NAME_2, divideUpstream);
-        assertEquals(1, upstreamMap.get(MOCK_SELECTOR_NAME_2).size());
-        assertEquals("localhost", upstreamMap.get(MOCK_SELECTOR_NAME_2).get(0).getUpstreamHost());
+        upstreamMap.put(MOCK_SELECTOR_NAME_2, new CopyOnWriteArrayList<>(Collections.singletonList(divideUpstream)));
         upstreamCheckService.replace(MOCK_SELECTOR_NAME_2, Collections.singletonList(divideUpstream2));
         assertEquals(1, upstreamMap.get(MOCK_SELECTOR_NAME_2).size());
         assertEquals("localhost2", upstreamMap.get(MOCK_SELECTOR_NAME_2).get(0).getUpstreamHost());

--- a/shenyu-admin/src/test/java/org/apache/shenyu/admin/service/UpstreamCheckServiceTest.java
+++ b/shenyu-admin/src/test/java/org/apache/shenyu/admin/service/UpstreamCheckServiceTest.java
@@ -221,8 +221,8 @@ public final class UpstreamCheckServiceTest {
         final DivideUpstream divideUpstream2 = DivideUpstream.builder()
                 .upstreamHost("localhost2")
                 .build();
-        upstreamMap.remove(MOCK_SELECTOR_NAME_2);
-        upstreamMap.put(MOCK_SELECTOR_NAME_2, new CopyOnWriteArrayList<>(Collections.singletonList(divideUpstream)));
+        UpstreamCheckService.removeByKey(MOCK_SELECTOR_NAME_2);
+        upstreamCheckService.replace(MOCK_SELECTOR_NAME_2, new CopyOnWriteArrayList<>(Collections.singletonList(divideUpstream)));
         upstreamCheckService.replace(MOCK_SELECTOR_NAME_2, new CopyOnWriteArrayList<>(Collections.singletonList(divideUpstream2)));
         assertEquals(1, upstreamMap.get(MOCK_SELECTOR_NAME_2).size());
         assertEquals("localhost2", upstreamMap.get(MOCK_SELECTOR_NAME_2).get(0).getUpstreamHost());

